src/Data/BitVector/Sized/Unsigned.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE TypeApplications #-}
+
+{-|
+Module      : Data.BitVector.Sized.Signed
+Copyright   : (c) Galois Inc. 2018
+License     : BSD-3
+Maintainer  : benselfridge@galois.com
+Stability   : experimental
+Portability : portable
+
+This module defines a wrapper around the 'BV' type, 'UnsignedBV', with
+instances not provided by 'BV'.
+-}
+
+module Data.BitVector.Sized.Unsigned
+  ( UnsignedBV(..)
+  , mkUnsignedBV
+  , mapUnsignedBV
+  ) where
+
+import           Data.BitVector.Sized.Internal (BV(..), mkBV)
+import qualified Data.BitVector.Sized.Internal as BV
+import Data.Parameterized.NatRepr
+
+import Data.Bits (Bits(..), FiniteBits(..))
+import Data.Ix
+import GHC.Generics
+import GHC.TypeLits
+import Numeric.Natural
+
+-- | Signed bit vector.
+newtype UnsignedBV w = UnsignedBV { asBV :: BV w }
+  deriving (Generic, Show, Read, Eq, Ord)
+
+mkUnsignedBV :: NatRepr w -> Integer -> UnsignedBV w
+mkUnsignedBV w x = UnsignedBV (BV.mkBV w x)
+
+mapUnsignedBV :: (BV w -> BV w) -> UnsignedBV w -> UnsignedBV w
+mapUnsignedBV f (UnsignedBV bv) = UnsignedBV (f bv)
+
+liftUnary :: (BV w -> BV w)
+          -> UnsignedBV w
+          -> UnsignedBV w
+liftUnary op (UnsignedBV bv) = UnsignedBV (op bv)
+
+liftBinary :: (BV w -> BV w -> BV w)
+           -> UnsignedBV w
+           -> UnsignedBV w
+           -> UnsignedBV w
+liftBinary op (UnsignedBV bv1) (UnsignedBV bv2) = UnsignedBV (op bv1 bv2)
+
+liftBinaryInt :: (BV w -> Natural -> BV w)
+              -> UnsignedBV w
+              -> Int
+              -> UnsignedBV w
+liftBinaryInt op (UnsignedBV bv) i = UnsignedBV (op bv (fromIntegral i))
+
+instance KnownNat w => Bits (UnsignedBV w) where
+  (.&.)        = liftBinary BV.and
+  (.|.)        = liftBinary BV.or
+  xor          = liftBinary BV.xor
+  complement   = liftUnary (BV.complement knownNat)
+  shiftL       = liftBinaryInt (BV.shl knownNat)
+  shiftR       = liftBinaryInt BV.lshr
+  rotateL      = liftBinaryInt (BV.rotateL knownNat)
+  rotateR      = liftBinaryInt (BV.rotateR knownNat)
+  bitSize _    = widthVal (knownNat @w)
+  bitSizeMaybe _ = Just (widthVal (knownNat @w))
+  isSigned     = const False
+  testBit (UnsignedBV bv) ix = BV.testBit' (fromIntegral ix) bv
+  bit          = UnsignedBV . BV.bit' knownNat . fromIntegral
+  popCount (UnsignedBV bv) = fromIntegral (BV.asUnsigned (BV.popCount bv))
+
+instance KnownNat w => FiniteBits (UnsignedBV w) where
+  finiteBitSize _ = widthVal (knownNat @w)
+
+instance KnownNat w => Num (UnsignedBV w) where
+  (+)         = liftBinary (BV.add knownNat)
+  (*)         = liftBinary (BV.mul knownNat)
+  abs         = liftUnary (BV.abs knownNat)
+  signum      = const $ UnsignedBV (BV 0)
+  fromInteger = UnsignedBV . mkBV knownNat
+  -- in this case, negate just means "additive inverse"
+  negate      = liftUnary (BV.negate knownNat)
+
+checkInt :: NatRepr w -> Int -> Int
+checkInt w i | 0 <= i && i <= fromIntegral (maxUnsigned w) = i
+             | otherwise = error "bad argument"
```

Comment:
Panic? Or is this actually an `error`?

Reply:
I'm not sure. I changed it to panic for now just to mimic the rest of the library, but perhaps `error` is more appropriate?

Reply:
Let's not use `fromIntegral`.  If we need to compare a fixed integer size against an `Integer`, always cast up toward integer to do the comparison.

Reply:
Meaning, specifically, use `toInteger` on `i` instead